src/components/picker/menu/index.tsx

```diff
@@ -1,111 +1,164 @@
-		if (!this.readonly && !this.multiple && event.detail.element.selected) {
-			for (const option of this.options.values())
-				if (option.element != event.detail.element)
-					option.element.selected = false
-		}
+	@Listen("smoothlyPickerOptionChange")
+	optionChangeHandler(event: CustomEvent<Option>) {
+		if (!this.listElement || !event.composedPath().includes(this.listElement)) {
+			event.stopPropagation()
+			this.options.get(event.detail.value)?.set.selected(event.detail.selected)
+		} else
+			this.backend.get(event.detail.value)?.set.selected(event.detail.selected)
+		if (!this.readonly && !this.multiple && event.detail.selected)
+			for (const option of chain(this.options.values(), this.backend.values()))
+				if (option.value != event.detail.value)
+					option.set.selected(false)
 	}
 	inputHandler(event: CustomEvent<Record<string, any>>) {
 		event.stopPropagation()
 		this.search = event.detail.search
 		if (!this.search) {
-			this.allowed = false
-			for (const option of this.options.values())
-				option.element.visible = true
+			this.valid = false
+			for (const option of chain(this.options.values(), this.backend.values()))
+				option.set.visible(true)
 		} else {
-			this.allowed = !Array.from(this.options.values()).find(option => option.value == this.search)
-			for (const option of this.options.values())
-				option.element.name.toLocaleLowerCase().includes(this.search.toLocaleLowerCase())
-					? (option.element.visible = true)
-					: (option.element.visible = false)
-		}
-	}
-	keyDownHandler(event: KeyboardEvent) {
-		if (event.key == "Enter") {
-			event.preventDefault()
-			this.addHandler()
+			this.valid = !Array.from(this.options.values()).find(option => option.value == this.search)
+			const search = this.search.toLocaleLowerCase()
+			for (const option of chain(this.options.values(), this.backend.values())) {
+				const value = option.value.toString().toLocaleLowerCase().includes(search)
+				const searches = option.search.some(value => value.includes(search))
+				const result = value || searches
+				option.set.visible(result)
+			}
 		}
 	}
 	addHandler() {
 		const validation = !this.validator ? true : this.validator(this.search)
 		if (typeof validation == "object" ? validation.result : validation) {
 			if (!this.multiple)
-				for (const option of this.options.values())
-					option.element.selected = false
-			this.new = [...this.new, { value: this.search, selected: true }]
+				for (const option of chain(this.options.values(), this.backend.values()))
+					option.set.selected(false)
+			this.created = new Map(this.created.set(this.search, { value: this.search, selected: true }).entries())
 			this.search = ""
-			this.searchElement?.focus()
+			this.valid = false
 		}
 		if (typeof validation == "object")
 			this.notice.emit(validation.notice)
 	}
+	keyDownHandler(event: KeyboardEvent) {
+		if (event.key == "Enter") {
+			event.preventDefault()
+			this.addHandler()
+		}
+	}
```

Comment:
Unused, in code. 
I think you might have removed it on the input.

Reply:
sure was by accident! its back now together with better focus handling that also was removed by accident.